banyand/metadata/embeddedserver/server.go

```diff
@@ -103,3 +139,37 @@ func NewService(_ context.Context) (metadata.Service, error) {
 	}
 	return s, nil
 }
+
+func (s *server) RegisterDefrag() {
+	var (
+		err        error
+		etcdLogger = logger.GetLogger().Named("etcd-server")
+		defrag     = func(_ time.Time, _ *logger.Logger) bool {
```

Comment:
Please test the defragmentation (defrag) function:

1. Extract the existing logic into a new, standalone function.
2. Create a new defrag function that takes two parameters: "listenURL" and "ecli."
3. Ensure that the function returns an error.

Next, start the embedded server to verify that the defrag function does not return any errors.